### ISSUE TITLE
i am using  'enablePrettyUrl' it working good for all url but error in dektrium/yii2-user

### ISSUE THREAD

**Issue by rahulsinh** [dektrium/yii2-user#351]
not change url of dektrium/yii2-user controllers give Not Found (#404) error

 $menuItems[] = ['label' => 'Signup', 'url' => ['/user/registration/register']];
 $menuItems[] = ['label' => 'Signin', 'url' => ['/user/security/login']];

**Comment by dmeroff** [comment]
Can't you only access those actions when `enablePrettyUrl` is true? Or you can't access actions even without `enablePrettyUrl` enabled?

**Comment by ZdamaSyi** [comment]
I had the same problem, with enablePrettyUrl disabled I was able to access actions.
front(back)end/config/main.php in
>'modules' => [
>        'user' => [
>            'class' => 'dektrium\user\Module',
>            ....
>            'urlRules' => [] //-- this helped me

**Comment by dougfelton** [comment]
I can confirm this issue, and I've been unable to find a fix.  I'm using the Basic Application Template (and therefore the Bootstrap Nav widget) and enablePrettyUrl set to true.  I've played around with urlRules for the module, as well as urlManager rules for the app, and have been unable to fix this issue sitewide.  The closest I came was adding this to my urlManager rules:

<module:user>/<controller:\w+>/<action:\w+>' => '<module>/<controller>/<action>

That fixes the problem for all non-user-module pages but then, on any pages of the user module, the URLs generated by the Nav widget begin looking like this:

/user/site/about (notice the extra 'user' at the beginning)

I tried making similar changes to urlRules in the user module, trying different combinations between that and the urlManager rules, but always end up with only ONE group of urls being correctly generated by the Nav widget: either the non-user-module URLs or the user-module URLs, but never both.

Any suggestions?

**Comment by dmeroff** [comment]
I still don't understand the problem, what is going wrong? You can't access user module's pages? Please answer following questions:

1) Which page do you want to access? Which url is generated using Url::to() for this route?
2) What is the problem? Which exception is thrown?

Also, please provide:
1) your urlManager component configuration
2) your yii2-user config (make sure you have `user` in modules section, but not in components)

**Comment by dougfelton** [comment]
I realize I didn't explain it very well.  I'll start over and attempt to simplify, so please ignore my last post.  To answer your questions:

1) My urlManager component configuration:

```
'<controller:\w+>/<id:\d+>' => '<controller>/view',
'<controller:\w+>/<action:\w+>/<id:\d+>' => '<controller>/<action>',
'<controller:\w+>/<action:\w+>' => '<controller>/<action>'
```

2) My yii-2 user config (user is only in modules section, not components):
```
'user' => [
     'class' => 'dektrium\user\Module',
     'admins' => ['myusername']
],
```

With the above configuration I am able to directly access any and every page of the website, including yii2-user admin pages.

The problem is that the generated links in my menu for login and logout are incorrect.  For example, I go to the home page, and clicking on the "Login" link gives me a 404 error, because the link is to '/user/login' instead of '/user/security/login'.

If instead of clicking on the Login link and getting a 404 error, I go directly to /user/security/login, I am able to access the page and log in without error.  Strangely though, on the login page, before logging in, I do notice that the links to '/site/index', '/site/about', '/site/contact' all appear with '/user' at the beginning ('/user/site/index', '/user/site/about', etc).

In any event, after I log in, I am successfully redirected to the home page, my '/site/about', '/site/index' nav links are now correct again, but my logout link is incorrect, reading '/user/logout' instead of '/user/security/logout'

Lastly, here is the the $items array for my nav widget, and as you can see, the login, logout, and register URLs are copied directly from the "Example of Menu" in the documentation.

```
'items' => [
	['label' => 'Home', 'url' => ['site/index']],
	['label' => 'About', 'url' => ['site/about']],
	['label' => 'Contact', 'url' => ['site/contact']],
	Yii::$app->user->isGuest ?
		['label' => 'Sign in', 'url' => ['/user/security/login']] :
		['label' => 'Sign out (' . Yii::$app->user->identity->username . ')',
			'url' => ['/user/security/logout'],
			'linkOptions' => ['data-method' => 'post']],
			['label' => 'Register', 'url' => ['/user/registration/register'], 'visible' => Yii::$app->user->isGuest]
],
```
I am using Yii 2.0.4 and the Basic Application Template.

**Comment by mzysk** [comment]
I am using same configuration as @dougfelton and same problem here. Login/Register is working great. Can't logout with properly generated post link (405 error).

My rules in common config:
```
'rules' => array(
                '<controller:\w+>/<id:\d+>' => '<controller>/view',
                '<controller:\w+>/<action:\w+>/<id:\d+>' => '<controller>/<action>',
                '<controller:\w+>/<action:\w+>' => '<controller>/<action>',
                '<alias:register>' => 'user/registration/<alias>',
                '<alias:logout|login>' => 'user/security/<alias>'
            ),
```

and logout button:
```
<?= Html::a(Yii::t('frontend', 'log out'), ['/user/security/logout'], ['class'=>'fr btn', 'data-method'=>'post']) ?>
```

**Comment by mzysk** [comment]
ok, last post from this thread helped me. Now it's working great.

http://www.yiiframework.com/forum/index.php/topic/60350-fresh-basic-yii-install-wont-log-out/

**Comment by dougfelton** [comment]
I've got it working now as well. mzysk's first post got me halfway there.  I added his last two rules (the <alias> rules) to my config file and that got me closer, and then I had to edit these three lines in my template:

```
['label' => 'Home', 'url' => ['site/index']],
['label' => 'About', 'url' => ['site/about']],
['label' => 'Contact', 'url' => ['site/contact']],
```

I had to add a / before the word 'site' so they now look like this:

```
['label' => 'Home', 'url' => [/'site/index']],
['label' => 'About', 'url' => ['/site/about']],
['label' => 'Contact', 'url' => ['/site/contact']],
```

After I did both of those things, everything worked as expected.  Note: those slashes I added are in the basic application template, so I must have removed them at some point and that was a large part of the problem... though not the entire problem.
